resource/systemd/states.go

```diff
@@ -66,13 +66,13 @@ const (
 	ASDeactivating ActiveState = "deactivating"
 )
 
-func (state ActiveState) Equal(state2 ActiveState) bool {
-	if state == state2 {
+func (state ActiveState) Equal(other ActiveState) bool {
+	if state == other {
 		return true
 	}
-	state = ActiveState(strings.Replace(string(state), "\"", "", -1))
-	state2 = ActiveState(strings.Replace(string(state2), "\"", "", -1))
-	return state == state2
+	stateStr := strings.Trim(string(state), "\"")
+	otherStr := strings.Trim(string(other), "\"")
+	return stateStr == otherStr
```

Comment:
This happens in the CheckProperty function when I get prop.Value.Value(). From my test It seems prop.Value.Value() returns "\"<property>\"", but prop.Value.String() returns "<property>". It also happens occasionally in the WaitForLoadedState function, where I get `unitStatus` in the select statement.